bookkeeper-server/src/main/java/org/apache/bookkeeper/client/LedgerMetadata.java

```diff
@@ -336,8 +340,9 @@ LedgerMetadataFormat buildProtoFormat() {
             builder.setCtime(ctime);
         }
 
-        if (hasPassword) {
-            builder.setDigestType(digestType).setPassword(ByteString.copyFrom(password));
```

Comment:
@sijie here, I was not sure why builder.setDigestType is kept in if block. I moved it out of this if block. I guess, it is the right thing to do.

Reply:
I think it was initially for BC, because we don't put any password and digesttype in metadata long time ago. so when it was introduced, password and digesttype are added at the same time. so that was the reason why digest is skipped. 

I am not sure what version introduced, it was probably 4.1.0 or 4.2.0. However I would suggest keeping the logic as same as before. 

```
if (hasPassword) {
   builder.setDigestType(digestType);
   if (withPassword) {
       ...
   }
}
```

we will have a separate issues to clean up these BC related legacy code when we enforce EOL of releases.